src/main/java/org/jenkinsci/plugins/workflow/cps/global/GroovyShellDecoratorImpl.java

```diff
@@ -29,6 +29,7 @@ public void configureShell(CpsFlowExecution context, GroovyShell shell) {
                 try {
                     shell.getClassLoader().addURL(new File(repo.workspace,"src").toURI().toURL());
                     shell.getClassLoader().addURL(new File(repo.workspace, UserDefinedGlobalVariable.PREFIX).toURI().toURL());
+                    shell.getClassLoader().addURL(new File(repo.workspace,"src/main/groovy").toURI().toURL());
```

Comment:
Because the `src` path includes this path, I was curious about what would happen if you imported a class in `src/main/groovy` twice, once using the class, and once using `main.groovy` as the package for the class. It looks like it fails with a `LinkageError`, which is safe, so I think that's fine, although a bit weird.

Here is a test case that fails with `java.lang.LinkageError: loader (instance of  org/jenkinsci/plugins/workflow/cps/CpsGroovyShell$CleanGroovyClassLoader): attempted  duplicate class definition for name: "Foo"` demonstrating what I mean:

```
@Test public void srcClassViaMultiplePaths() throws Exception {
    sampleRepo.init();
    sampleRepo.write("src/main/groovy/Foo.groovy", "class Foo { }");
    sampleRepo.git("add", "src");
    sampleRepo.git("commit", "--message=init");
    GlobalLibraries.get().setLibraries(Collections.singletonList(
        new LibraryConfiguration("lib",
            new SCMSourceRetriever(new GitSCMSource(null, sampleRepo.toString(), "", "*", "", true)))));
    WorkflowJob p = r.jenkins.createProject(WorkflowJob.class, "p");
    p.setDefinition(new CpsFlowDefinition(
            "@Library('lib@master')\n +
            "import Foo\n" +
            "new main.groovy.Foo()", true));
    r.buildAndAssertSuccess(p);
}
```